security/pkg/k8s/controller/workloadsecret.go

```diff
@@ -399,7 +411,26 @@ func (sc *SecretController) scrtDeleted(obj interface{}) {
 
 func (sc *SecretController) namespaceUpdated(oldObj, newObj interface{}) {
 	oldNs := oldObj.(*v1.Namespace)
-	newNs := newObj.(*v1.Namespace)
+	newNs, ok := newObj.(*v1.Namespace)
```

Comment:
will this run in istiod or only citadel?

Reply:
It is in Citael and the calling flow is: runCA() -> sc=NewSecretController() -> sc.Run().

Reply:
Update: per offline discussion, the controller has been moved to istiod.